gui/src-tauri/src/main.rs

```diff
@@ -154,6 +214,58 @@ fn main() {
 			Ok(())
 		})
 		//
-		.run(tauri::generate_context!())
-		.expect("error while running tauri application");
+		.run(tauri::generate_context!());
+	match res {
+		Ok(()) => {}
+		Err(_) => res.expect("error while running tauri application"),
+	}
+}
+
+fn valid_java_paths() -> Vec<(OsString, i32)> {
+	let mut file = Builder::new()
+		.suffix(".class")
+		.tempfile()
+		.expect("Couldn't generate .class file");
+	file.write_all(include_bytes!("JavaVersion.class"))
+		.expect("Couldn't write to .class file");
+	let java_version = file.into_temp_path();
+
+	// Check if main Java is a supported version
+	let main_java = if let Ok(java_home) = std::env::var("JAVA_HOME") {
```

Comment:
I can't seem to find a way to implement ``let-else`` statements, I see that they are mostly used for early returns.

In this case how can I convert the ``JAVA_HOME`` var to an ``OsString``?
I could use ``.map`` but we are mixing functional with expressions again.

Then the other ``if let`` is okay as it is because it doesn't use an ``else``

Reply:
Looks like I misunderstood how let else works. The else must return `!`. So you're right, a let else is not possible here.